libbeat/template/processor.go

```diff
@@ -317,11 +317,10 @@ func (p *Processor) wildcard(f *mapping.Field, analyzers common.MapStr) common.M
 
 	property["type"] = "wildcard"
 
-	switch f.IgnoreAbove {
-	case 0: // Use libbeat default
-		property["ignore_above"] = defaultIgnoreAbove
-	case -1: // Use ES default
-	default: // Use user value
+	/* For wildcard fields, unlike keyword, don't force a default ignore_above limit.
+	   The default in ES will be used unless an explicit limit is set.
+	*/
```

Comment:
Could you add some additional context for why this is the case? There is some good (and convincing) information in https://github.com/elastic/beats/issues/30096, specifically:

> A big benefit wildcard has over keyword is the indexing of strings larger than the 32,766 byte Lucene max, but users can't gain that benefit if ignore_above is set to ignore string values greater than 1024 chars.